fuzz/fuzz_targets/process.rs

```diff
@@ -0,0 +1,80 @@
+// The fuzzing harness fuzz test some of the the
+// unicode string normalization processing
+
+#![no_main]
+
+#[macro_use]
+extern crate libfuzzer_sys;
+extern crate unicode_normalization;
+
+use unicode_normalization::{
+    char::{
+        canonical_combining_class, compose, decompose_canonical, decompose_compatible,
+        is_combining_mark,
+    },
+    UnicodeNormalization,
+};
+
+fuzz_target!(|data: &[u8]| {
+    let mut data = data;
+    let c = if let Some((char_value, remaining_data)) = data.split_first() {
+        match std::char::from_u32(*char_value as u32) {
+            Some(ch) => {
+                data = remaining_data;
+                ch
+            }
+            None => return,
+        }
+    } else {
+        return;
+    };
+
+    // Generate second character for fuzzing if data is enough
+    let c2 = if let Some((char_value, remaining_data)) = data.split_first() {
+        data = remaining_data;
+        std::char::from_u32(*char_value as u32)
+    } else {
+        None
+    };
+    let string_data: String = data
+        .iter()
+        .filter_map(|&b| std::char::from_u32(b as u32))
+        .collect();
```

Comment:
is there a reason you're manually constructing a string instead of having the data type be `(u8, u8, &str)`? The fuzzing infrastructure supports that.

Reply:
and probably an additional integer for picking the function type. you could also use an enum and `#[derive(Arbitrary)]` on it.

Reply:
Actually, I'd have the data be `(u8, &str)`, or `(NormalizationFunctionKind, &str)` and then in the `compose()` case pull the first character off of the string.

Reply:
Thanks for your comments and suggestions. I have changed how the string, characters and random index being retrieved. Now it should be more simplified and clear.